opm/autodiff/BlackoilModelBase.hpp

```diff
@@ -175,7 +176,7 @@ namespace Opm {
         /// \param[in, out] reservoir_state   reservoir state variables
         /// \param[in, out] well_state        well state variables
         template <class NonlinearSolverType>
-        IterationReport nonlinearIteration(const int iteration,
+        SimulatorReport nonlinearIteration(const int iteration,
```

Comment:
In the medium term I intend to do so. It would also be good for the fully implicit solvers to account for failed time steps; come think about it, that's probably the bulk of the "missing" solver time. the only obstacle for this which I can come up with of out of my head is that access to the report object must be made exception-proof for this.